crates/sui-sdk/examples/utils/example_utils.rs

```diff
@@ -0,0 +1,150 @@
+    let addresses = in_mem_keystore.addresses();
+
+    let second_address = addresses
+        .get(0)
+        .expect("Wasn't able to generate a Sui address in-memory. Aborting.");
+
+    Ok(vec![active_address, *second_address])
+}
+
+/// Request tokens from the Faucet for the given address
+pub(crate) async fn request_tokens_from_faucet(address: SuiAddress) -> Result<(), anyhow::Error> {
+    let address_str = address.to_string();
+    let json_body = json![{
+        "FixedAmountRequest": {
+            "recipient": &address_str
+        }
+    }];
+
+    // make the request to the faucet JSON RPC API for coin
+    let client = Client::new();
+    let resp = client
+        .post(SUI_FAUCET)
+        .header("Content-Type", "application/json")
+        .json(&json_body)
+        .send()
+        .await?;
+    println!("Faucet request status: {}", resp.status());
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+/// This function returns a transaction digest from a split coin transaction
+pub(crate) async fn split_coin_digest(sui: &SuiClient) -> Result<TransactionDigest, anyhow::Error> {
+    let addresses = local_wallet_addresses(sui).await?;
+    let sender = addresses
+        .get(0)
+        .expect("Keystore did not manage to generate two keys in memory. Aborting");
+
+    // Search for the coins in the sender's address
+    let coins = sui
+        .coin_read_api()
+        .get_coins(*sender, None, None, None)
+        .await?;
+    let balance = sui
+        .coin_read_api()
+        .get_balance(*sender, Some("0x2::sui::SUI".to_string()))
+        .await?;
+
+    if coins.next_cursor.is_none() || balance.total_balance < 1 {
+        // Add coins to the newly created sender_address
+        request_tokens_from_faucet(*sender).await?;
+    }
+
+    println!(
+        "Address {sender} has {} coins ({:?}) and a total balance of {}\n",
+        coins.data.len(),
+        coins.next_cursor.unwrap(),
+        balance.total_balance
+    );
+
+    let Some(coin) = sui
+        .coin_read_api()
+        .get_coins(*sender, None, None, None)
+        .await?
+        .next_cursor
+        else {
+            panic!("Faucet did not work correctly and the provided Sui address has no coins")
+        };
+
+    let txb = sui.transaction_builder();
+    println!("Trying to split coin");
+    let txb_res = txb
+        .split_coin(*sender, coin, vec![1], None, 5_000_000)
+        .await?;
+
+    let keystore = FileBasedKeystore::new(&sui_config_dir()?.join(SUI_KEYSTORE_FILENAME))?;
+    let signature = keystore.sign_secure(sender, &txb_res, Intent::sui_transaction())?;
+
+    let transaction_response = sui
+        .quorum_driver_api()
+        .execute_transaction_block(
+            Transaction::from_data(txb_res, Intent::sui_transaction(), vec![signature]),
+            SuiTransactionBlockResponseOptions::full_content(),
+            Some(ExecuteTransactionRequestType::WaitForLocalExecution),
```

Comment:
If you only want the digest of the transaction, you do not need to request `full_content`.  You can use `SuiTransactionBlockResponseOptions::new()` (that will tell the RPC's response to contain just the digest).